usr.bin/paste/paste.c

```diff
@@ -98,51 +114,50 @@ main(int argc, char *argv[])
 		delim = tab;
 	}
 
-	if (seq)
-		rval = sequential(argv);
-	else
-		rval = parallel(argv);
+	for (filecnt = 0; *argv; ++argv, ++filecnt) {
+		failed = 0;
+		if ((lp = malloc(sizeof(LIST))) == NULL)
+			err(EXIT_FAILURE, NULL);
+		if ((*argv)[0] == '-' && !(*argv)[1])
+			lp->fp = stdin;
+		else if (!(lp->fp = fopen(*argv, "r"))) {
+			if (!seq)
+				err(EXIT_FAILURE, "%s", *argv);
+			else
+				failed = errno;
+		}
+
+		if (cap_rights_limit(fileno(lp->fp),
+		    cap_rights_init(&rights, CAP_READ)) < 0 &&
+		    errno != ENOSYS)
+			err(EXIT_FAILURE, "unable to limit rights for %s", *argv);
+
+		lp->cnt = filecnt;
+		lp->err = failed;
+		lp->name = *argv;
+
+		STAILQ_INSERT_TAIL(&lh, lp, entries);
+	}
+
+	if (cap_enter() < 0 && errno != ENOSYS)
```

Comment:
Why not use `caph_enter`?

Reply:
I chose to use the raw Capsicum API instead of `caph` because it felt more concise and appropriate for the small-scale use cases in such a utility. Additionally, I thought having another direct example of capsicumisation with only raw API calls might be beneficial as a reference, considering further complexities observed here. While it does add some verbosity to the code, I don't see it causing any other significant issues. That said, if you still believe using Capsicum helpers would be better for this context, I’m happy to reconsider and follow your guidance.